src/history/base.rs

```diff
@@ -322,4 +324,14 @@ mod test {
 
         Ok(())
     }
+
+    #[test]
+    fn clear_history() -> Result<()> {
+        let mut history = create_filled_example_history()?;
+        assert_ne!(history.count_all()?, 0);
+        history.clear().unwrap();
+        assert_eq!(history.count_all()?, 0);
```

Comment:
Maybe we should drop it and try to reopen it, to exclude a potential error case like we were seeing with nushell's sqlite "clear"

Reply:
Hmm, that's a good idea but it's a little tricky because `create_filled_example_history()` doesn't actually create files. It creates an in-memory database for SQLite and never sets the filename for the text history.

It would be a good idea to have tests that exercise filesystem functionality though 🤔

Reply:
I thought we had such tests but maybe that was before the sqlidation or are only on the individual implementations.

Reply:
@sholderbach I've added a test that uses a file-backed history.

Reply:
omg the reedline CI is so fast compared to Nushell 😭